core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -58,6 +52,23 @@ protected DataTableScan(TableOperations ops, Table table, Long snapshotId, Schem
     super(ops, table, snapshotId, schema, rowFilter, caseSensitive, colStats, selectedColumns, options);
   }
 
+  @Override
+  public TableScan appendsBetween(Long fromSnapshotId, Long toSnapshotId) {
+    final Long scanSnapshotId = snapshotId();
+    Preconditions.checkState(scanSnapshotId == null,
+        "Cannot enable incremental scan, scan-snapshot set to id=%s", scanSnapshotId);
+    return new IncrementalDataTableScan(
+        tableOps(), table(), schema(), filter(), isCaseSensitive(), colStats(), selectedColumns(), options(),
+        fromSnapshotId, toSnapshotId);
+  }
+
+  @Override
+  public TableScan appendsAfter(Long fromSnapshotId) {
+    final Snapshot currentSnapshot = table().currentSnapshot();
+    Preconditions.checkState(currentSnapshot != null, "Current snapshot is null");
```

Comment:
Can we improve this error message? It is accurate, but it doesn't explain why that's a problem. How about "Cannot scan appends after %s, there is no current snapshot"?